client-js/app/elements/labrad-plot.ts

```diff
@@ -498,11 +514,10 @@ export class Plot extends polymer.Base {
         }
       }
 
-      for (let i = 2; i < row.length; i++) {
-        let z = row[i];
+        let z = row[this.displayTraces[0]+2];
```

Comment:
pull out `this.displayTraces[0] + 2` into a named variable and use it here and below for more DRYness.